packages/wallet/test/hardware/ledger/LedgerKeyAgent.test.ts

```diff
@@ -219,6 +219,53 @@ describe('LedgerKeyAgent', () => {
     });
   });
 
+  describe('deviceConnections persistence', () => {
+    let transportSpy: jest.SpyInstance;
+    beforeAll(async () => {
+      transportSpy = jest.spyOn(LedgerKeyAgent, 'createTransport');
+
+      if (keyAgent.deviceConnection) {
+        await keyAgent.deviceConnection.transport.close();
+      }
+    });
+
+    afterEach(() => {
+      transportSpy.mockClear();
+    });
+
+    it('should return an existing connection if one exists', async () => {
+      LedgerKeyAgent.deviceConnections = [];
+      const connection1 = await LedgerKeyAgent.establishDeviceConnection(CommunicationType.Node);
+      const connection2 = await LedgerKeyAgent.establishDeviceConnection(CommunicationType.Node);
+
+      expect(transportSpy).toHaveBeenCalledTimes(1);
+
+      await connection1.transport.close();
```

Comment:
It should be the same object so we don't need to close both, would be better to have an assertion like this instead
```suggestion
      expect(connection1).toBe(connection2);
```